rosbag2_transport/test/rosbag2_transport/test_play_timing.cpp

```diff
@@ -164,19 +164,34 @@ TEST_F(PlayerTestFixture, playing_rate_negative)
 
 TEST_F(PlayerTestFixture, playing_respects_delay)
 {
-  rclcpp::Duration delay_margin(0.5s);
+  rclcpp::Duration upper_bound_tolerance(1s);
+  rclcpp::Duration lower_bound_tolerance(2us);
+
+  // Sleep 5.0 seconds before play
+  play_options_.delay = rclcpp::Duration(5, 0);
+  // Don't expect accuracy more than a few microseconds
+  auto lower_expected_duration = (message_time_difference + play_options_.delay -
+    lower_bound_tolerance).to_chrono<std::chrono::microseconds>();
+  auto upper_expected_duration = (message_time_difference + play_options_.delay +
+    upper_bound_tolerance).to_chrono<std::chrono::microseconds>();
+
+  // Sanity check for steady clock accuracy
+  ASSERT_TRUE(std::chrono::duration_cast<std::chrono::microseconds>(
+      typename std::chrono::steady_clock::duration(1)).count() <= 1) <<
+    "The C++ compiler doesn't provide a precise enough steady_clock. The steady_clock duration"
+    " shall have a resolution of at least 1 microsecond.";
+  static_assert(std::ratio_less_equal_v<std::chrono::steady_clock::period, std::micro>,
+    "The C++ compiler doesn't provide a precise enough steady_clock. The steady_clock period shall"
+    " be at least 1 microsecond.");
```

Comment:
Another comment: should the test simply be skipped in that case? For platforms with a clock that isn't precise enough, this test will just fail and nothing can be done.

Reply:
Thanks for the review. I think an explicit test failure will be better than just automatically skipping it.  
It is better to adjust the skip manually if the test will fail due to insufficient precision in the clock.